File: tasks/lectures/jnu/java-ai-basic/task-2/java/6.java
import java.lang.Math;

public class Main {
  public static void main(String[] args) {
    // 배열 초기화
    int[][] arr = new int[4][4];
    // 이중 포문으로 배열의 각 값 접근
    for (int i = 0; i < 4; i++) {
      for (int j = 0; j < 4; j++) {
        arr[i][j] = (int)(Math.random()*10+1); // 문제 제시 값
      }
    }
    for (int i = 0; i < 4; i++) {
      for (int j = 0; j < 4; j++) {
        // 행 출력
        System.out.print(String.format("%d ", arr[i][j]));
      }
      // 하나의 열을 모두 출력하면 개행
      System.out.println();
    }
  }
}

File: tasks/lectures/jnu/c-ce-basic-kim-2022-1/l-4.25/6.c
// 포인터 배열
// -> 배열과 포인터

#include <stdio.h>

int main() {
  int arr[5] = {10, 20, 30, 40, 50};

  for (int i = 0; i < 5; i++) {
    printf("&arr[%d] = %p,\t", i, &arr[i]);
    printf("arr+%d = %p\n", i, arr + i);
  }

  for (int i = 0; i < 5; i++) {
    // 왼쪽과 오른쪽의 문법이 같음: 배열은 포인터로 관리되므로
    printf("arr[%d] = %p,\t", i, arr[i]);
    printf("*(arr+%d) = %p\n", i, *(arr + i));
    // *(arr+4) = arr[4]
  }

  // 포인터를 사용하면 변수의 스코프를 무시할 수 있음
  // => 간편하나 위험
  return 0;
}